packages/gatsby/src/state-machines/develop/index.ts

```diff
@@ -101,6 +101,9 @@ const developConfig: MachineConfig<IBuildContext, any, AnyEventObject> = {
         SOURCE_FILE_CHANGED: {
           actions: [forwardTo(`run-queries`), `markSourceFilesDirty`],
         },
+        ADD_NODE_MUTATION: {
+          actions: [forwardTo(`run-queries`)],
+        },
```

Comment:
Rather than forwarding it to the child machine, it should run the action in the current machine. See the `initializingData` state for an example.

Reply:
The trouble here is that this is not supposed to run them immediately: the reason we do this deferral is so that nodes _aren't_ mutated during query runs. Changing this would mean that we'd be back to how it was without any deferral. Now that might be what we want to do after all, but at that point it would be better to remove the whole deferral system. 

I think we need a different way of solving this though, which is probably a more granular way of dealing with deferred resolver mutations, i.e. not waiting until `waiting`, but bailing and drainign earlier.

Reply:
Yeah, that's what I thought. This was just a quick and direct attempt to address this problem but I see that it defeats the whole purpose of node deferring. We can definitely have a call/chat tomorrow as I'd like to get more context (haha) on our state charts in general. CC @wardpeet